### PR TITLE
feat: implement additional variables and `bldr eval`

### PR DESCRIPTION
**Pull request by smira**
The goal of this PR is to enable custom image tags for extensions.

This PR adds deep nested templated additional variables per package
(more precisely, per directory) `vars.yaml` which might be used to
template `pkg.yaml`.

New command `bldr eval` can be used to output value of any variable.

Build arguments are injected as `BUILD_ARG_` variables.

The idea is to declare extension version as a variable, and use that
variable to produce image tag and in `pkg.yaml`.

Signed-off-by: Andrey Smirnov <andrey.smirnov@talos-systems.com>

### CODE REVIEW

File: internal/pkg/integration/testdata/variables/test.yaml [hunk 1]
```diff
@@ -27,3 +27,17 @@ run:
   - name: validate
     runner: validate
     expect: success
+  - name: eval-local
+    runner: eval
+    platform: linux/amd64
```
Comment (frezbo) [inline internal/pkg/integration/testdata/variables/test.yaml:32]: any reason explicit platform was specified, or does this not work with arm64?

Comment (smira) [reply internal/pkg/integration/testdata/variables/test.yaml:32]: good question, I think it is even ignored actually or not used, I will double check and remove it

**Comment by smira** [comment]
See https://github.com/siderolabs/extensions/pull/26 as an example of how this can be used

File: internal/pkg/pkgfile/build.go [hunk 2]
```diff
@@ -143,7 +147,7 @@ func Build(ctx context.Context, c client.Client, options *environment.Options) (
 				Definition: def.ToPB(),
 			})
 			if err != nil {
-				return fmt.Errorf("failed to resolve dockerfile: %q", err)
+				return fmt.Errorf("failed to resolve dockerfile: %w", err)
```
Comment (frezbo) [inline internal/pkg/pkgfile/build.go:150]:
```suggestion
				return fmt.Errorf("failed to resolve %s: %w", constants.PkgFile, err)
```

**Review by frezbo — APPROVED** [review]

File: internal/pkg/pkgfile/build.go (hunk 2 above)
Comment (smira) [reply internal/pkg/pkgfile/build.go:150]: it looks like this error is actually about solving the whole LLB, so it was completely inaccurate

**Pull request merged**